### ISSUE TITLE
[BUG] Missing/Broken @mswjs files

### ISSUE THREAD

**Issue by SomervilleTom** [markerikson/redux-quickstart-example-app#62]
I'm seeing 28 warnings, in two groups of 14 each, pertaining to `@mswjs` each time I start the `redux-essentials-example` app using `yarn start`. I'm pretty sure this is, in turn, causing the app itself to crash while using `https`. I run all this on a robust AWS EC2 instance running a recent version Rocky Linux.

I'm using version of the tutorial forked from this repo:
```
git clone git@github.com:reduxjs/redux-essentials-example-app.git
```

I've already worked around the known OpenSSL issue. I run multiple other apps on the same platform using https with no complaints -- those each use `axios` for all interactions between the browser and server and there are no issues with installed certs.

Here are the complaints I see at the command line (an SSH client) when I start the app with `vscode' (my IDE) using `launch.json`:
```
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/createInterceptor.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/createInterceptor.ts'

Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/index.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/index.ts'

Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts'

Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/index.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/index.ts'

Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/polyfills/EventPolyfill.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/polyfills/EventPolyfill.ts'

Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/polyfills/ProgressEventPolyfill.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/polyfills/ProgressEventPolyfill.ts'

Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/utils/bufferFrom.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/utils/bufferFrom.ts'

Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/utils/createEvent.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/utils/createEvent.ts'

Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/fetch/index.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/fetch/index.ts'

Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/remote.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/remote.ts'

Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/getCleanUrl.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/getCleanUrl.ts'

Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/parseJson.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/parseJson.ts'

Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/toIsoResponse.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/toIsoResponse.ts'

Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/uuid.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/uuid.ts'

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/createInterceptor.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/createInterceptor.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/createInterceptor.ts'

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/index.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/index.ts'

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/interceptors/XMLHttpRequest/XMLHttpRequestOverride.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts'

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/interceptors/XMLHttpRequest/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/index.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/index.ts'

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/interceptors/XMLHttpRequest/polyfills/EventPolyfill.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/polyfills/EventPolyfill.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/polyfills/EventPolyfill.ts'

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/interceptors/XMLHttpRequest/polyfills/ProgressEventPolyfill.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/polyfills/ProgressEventPolyfill.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/polyfills/ProgressEventPolyfill.ts'

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/interceptors/XMLHttpRequest/utils/bufferFrom.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/utils/bufferFrom.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/utils/bufferFrom.ts'

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/interceptors/XMLHttpRequest/utils/createEvent.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/utils/createEvent.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/XMLHttpRequest/utils/createEvent.ts'

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/interceptors/fetch/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/fetch/index.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/interceptors/fetch/index.ts'

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/remote.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/remote.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/remote.ts'

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/utils/getCleanUrl.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/getCleanUrl.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/getCleanUrl.ts'

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/utils/parseJson.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/parseJson.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/parseJson.ts'

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/utils/toIsoResponse.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/toIsoResponse.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/toIsoResponse.ts'

WARNING in ./node_modules/msw/node_modules/@mswjs/interceptors/lib/utils/uuid.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/uuid.ts' file: Error: ENOENT: no such file or directory, open '/home/tms/redux_play/essentials/node_modules/msw/node_modules/@mswjs/interceptors/src/utils/uuid.ts'

webpack compiled with 14 warnings
```

Here is the (Chrome) console complaint when I attempt to launch the app using `https`:

```
Uncaught Error Error: [MSW] Failed to register the Service Worker:

Failed to register a ServiceWorker for scope ('https://localhost:5099/') with script ('https://localhost:5099/mockServiceWorker.js'): An SSL certificate error occurred when fetching the script.
    at <anonymous> (/home/tms/redux_play/essentials/node_modules/msw/lib/esm/index.js:670:1)
    at fulfilled (/home/tms/redux_play/essentials/node_modules/msw/lib/esm/RequestHandler-deps.js:34:1)
    --- await ---
    at ./src/index.js (/home/tms/redux_play/essentials/src/index.js:25:1)
    at options.factory (/home/tms/redux_play/essentials/webpack/runtime/react refresh:6:1)
    at __webpack_require__ (/home/tms/redux_play/essentials/webpack/bootstrap:24:1)
    at <anonymous> (/home/tms/redux_play/essentials/webpack/startup:7:1)
    at <anonymous> (localhost꞉5099/static/js/bundle.js:165010:12)
```

I note that file generating the browser complaints (`public/mockServiceWorker.js`) is in the github repo. I'm using it unchanged.

These warnings are in two groups of 14 complaints each, and each appears to have a similar pattern.

1. Incorrect import path in each of the first 14 complaints

    The path reported in the first of each 14 lines of the complaint is:

    ```
   node_modules/msw/node_modules/@mswjs/interceptors/src/<filename>.ts
    ```

   My `node_modules` has no 'src' directory. It has a 'lib' directory that contains the following:

    ```
    createInterceptor.d.ts
    createInterceptor.js
    createInterceptor.js.map
    index.d.ts
    index.js
    index.js.map
    remote.d.ts
    remote.js
    remote.js.map
    ```

2. Incorrect source path in each of the second group

    The path reported in each file of the second group of 14 complains is the same as above:

    ```
    node_modules/msw/node_modules/@mswjs/interceptors/src/<filename>.ts
    ```

    I used "find" to search for occurrences of the first of the 15 filenames ('createInterceptor.ts') and found none.

Here are the 14 missing files:
```
'createInterceptor.ts'
'index.ts'
'interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts'
'interceptors/XMLHttpRequest/index.ts'
'interceptors/XMLHttpRequest/polyfills/EventPolyfill.ts'
'interceptors/XMLHttpRequest/polyfills/ProgressEventPolyfill.ts'
'interceptors/XMLHttpRequest/utils/bufferFrom.ts'
'interceptors/XMLHttpRequest/utils/createEvent.ts'
'interceptors/fetch/index.ts'
'remote.ts'
'utils/getCleanUrl.ts'
'utils/parseJson.ts'
'utils/toIsoResponse.ts'
'utils/uuid.ts'
```

I haven't worked through the tutorial, and so I don't have any insight into how serious this is. I very much prefer to provide only an https listener on each open port.

I appreciate your attention, and even more so I appreciate all the good and hard work already invested in this excellent product and tutorial.

**Comment by markerikson** [comment]
Not sure what's causing the `source-map-loader` warnings, but those shouldn't affect any runtime behavior.

The actual error seems to be some kind of HTTPS/SSL certificate mismatch, but not sure what.

I _will_ say that the example app _should_ be running on `http://localhost:3000` by default, _not_ `https` or `:5099`.

**Comment by SomervilleTom** [comment]
I found and fixed the issue(s). As it turns out, there were two -- the first in `msw` and the second in my code.

It turns out that this was a [bug in `msw`](https://github.com/mswjs/msw/issues/1070) that was fixed in release 0.38.0. I edited `package.json` so that the entry for `msw` is as follows:
```
"msw": "^0.38.0",
```

After making this change, I ran "yarn install" from the command line. The openSSL issue continues to plague CRA, and so I had to apply the usual patch (there is validation code in `react-scripts/config/getHttpsConfig.js` that has to be modified). With these changes in place, the tutorial app launches cleanly and expects to be connected using an https URL.

I then edited my `launch.json` to use the actual URL of my server instead of localhost. This is needed because the server has an actual certificate and the browser correctly complains about connecting with "localhost" or any of its variants.

With these changes, the "essentials" opens and runs as described.

This is an excellent package and tutorial, and I really appreciate the hard work invested in creating and documenting both the packages and the tutorial. It will CERTAINLY improve the code of my project.
